### DISCUSSION TITLE
Sane number of loot items?

### DISCUSSION THREAD

**Discussion by Danimal696** [flareteam/flare-game#905] [Q&A]
Hello again, I want advice from you guys, now I can make randomized items by the hundreds, but what would be a good number per quality level (which is the item lv)?. The total list of possible items is this:

- Armor warrior/mage: Head, chest, gloves, belts, pants, shoes
- Weapons: Axes, swords, clubs, books , ranged, staffs, wands, shields
- Others: rings, pendants

And they will disput their place with others by having some special default advantage, like daggers being faster and having some +crit, clubs  having stun chance (is it possible?), axes +damage... So the progression it´s not as marked as going dagger>shortword>Sword>Greatsword, now you have several weapons options per "weapon tier" (same for armors). 

Qualities are marked by number of bonus an item possess, from normal (no bonus), magic (1-2), rare (3-4), unique (5 or more), so I would need to create items that have bonus going from 1 to 5 for each item level. 

Meaning i need items for each lv * quality * type, in Diablo the generator just spits an item the same lv as the enemy, but here it´s not possible; meaning the loot list can get huge, I will use the same cheat as before where enemies drops items from a list that will be used throught a whole act (maybe use 2 or 3 higher level loots lists per act if needed) the needed stats to use some item are what will stop the player to use an item too powerful for their level.

See how it goes out of hand?, what do you think would be a good number for each item? item drop chance is also a bit wonky, just a 5% increase makes the item drop way more than the others, so if normal items had a bigger drop chance than the others you would barely see anything else but "normies". Help, please?

**Comment by WithinAmnesia** [comment]
How many item levels do you have / are planning to have? If it is around ~50 or less this can work. If you have like 300 or 500 or more than just change the level numbers to fit the ratios.

At least one item of similar type and or class archetype role (melee one hand for example can be axe, dagger, sword etc. or a caster chest like cloth or leather mail plate etc.) per questing quality range (as in like regular stuff a player could see questing, not ~'legendary' items etc.). Try to not have big 'gaps' where a player must 'jump' from gear bracket to gear bracket too much / give a few / a handful of progression options per level bracket.

As like as a 20-30 level bracket could have like ~30 to ~50 / ~40 different 1H weapons with all different roles supported at least 2 or 3 times within the 20-30 level bracket. Say there are ~40 1H weapons, half could be 'melee' and the other half could be 'caster' and then look at each class / build type and type to not have more than a ~7 level gap for upgrade options for each role. Like a level level 18 caster dagger could have a level 20 caster mace or a level 23 caster sword for an upgrade. Or later on as a level 26 caster dagger or a level 29 caster mace or a level 30 caster sword etc. That is how vanilla wow tends to operate for 'rare' / dungeon loot / big upgrades.

Then almost every level has lesser stuff like uncommon / 'green' items of almost every type and kind at almost every item level (maybe in ~6-7 covered in 10 levels of say a green dagger). With often some green / 'easy' upgrades have many options / versions for the same item with random stats to cover many play style / class roles like: https://classic.wowhead.com/item=15278/solstice-staff (has melee and caster variants on the same green / uncommon / 'easy' questing / leveling / progression item).

Or basically look at this (it works well enough to cover a wide range of progression for many class / play-style roles: https://classic.wowhead.com/items/weapons/type:6:10:1:5:8 for how the items are distributed. Notice that roughly all questing levels have a wider range of possible options for different class archetype / role / play style build stats as well. Like some are for damage strength attack power agility crit hit chance on hit etc. some are for ~casters or druids with them being a ~stat stick with intellect, spell / healing power, spell crit, spell hit, spirit, mana reg etc.

Or you could also look at Diablo II and kind in mind the above comments as some points to look out for and try something like this https://diablo2.io/database/

**Comment by Danimal696** [comment]
The thing is each level should have XX number of each item, in all it´s qualities. So, say, my first act ends at lv12; that means all 12 levels will need to have items of all qualities (common, magical...) and all types (weapons, armors). So for sanity´s shake it should not be an endless list, but make it shorth and items really start repeating, perhaps I should just make a huge list and sent it all to hell.

**Reply by WithinAmnesia** [reply]
@Danimal696 Well first off, figure out how many equip-able item types, class / player style archetypes you have, then how many levels you have and what 'end game' should / could look like. Try to find your more or less reasonable limits and then find ways to bridge the gaps. I guess you have around 3-5 acts planned out? so 12 x 3 = 36, 12 x 5 = 60, 36 + 60 = 96 / 2 = 48, so your max level could be around ~50?

I guess you have cloth, leather, mail and plate armours and all Flare weapon types with roughly melee, ranged and caster without a dedicated healer build or any type of group / instance content specs (because single player engine). I'm guessing You at least want common, uncommon, rare, epic and legendary quality items. Possibly with common being +/- 0 items levels per ~required level, uncommon being ~+5 item level, rare being ~+10 item level, epic being ~+15 item level and endgame / requiring level 50 legendary items being ~+20 item levels or very close to around that kind of idea.

If we have 3 main play styles for damage with melee, ranged and caster / ~mage we can in general get at least 3 core or 'pure-ish build' style weapons and armour.

I'll use 1 to 50 item levels to get an idea going.

Item Level Brackets (^~Max Character Level ~50; *End Game with optional ~'coin pusher' / high difficulty content, dungeons, bosses etc.)
1-5 common all weapon slots, cloth, leather, mail (maybe plate) armour in few slots, uncommon of every 'major slot', no rares and up.
6-10 common all weapon slots, cloth, leather, mail (maybe plate) armour in some slots, uncommon of some slots, no rares and up.
11-15 common all weapon slots, cloth, leather, mail (maybe plate) armour in many slots, uncommon of most slots, handful of rares only.
16-20 common all weapon slots, cloth, leather, mail, plate armour in all slots, uncommon of all slots, some rares only.
21-25 common all weapon slots, cloth, leather, mail, plate armour in all slots, uncommon of all slots, some rares only.
26-30 common most slots with weapons, cloth, leather, mail with all slots plate armour, uncommon of all slots, many rares, few epics.
31-35 common most slots with weapons, cloth, leather, mail with all slots plate armour, uncommon of all slots, many rares, few epics.
36-40 common most slots with weapons, cloth, leather, mail, many slots plate armour, uncommon of all slots, many rares, some epics.
41-45 common some slots with weapons, cloth, leather, mail, plate armour, uncommon of all slots, rares of all slots, some epics.
46-50^ common some slots with weapons, cloth, leather, mail, plate armour, uncommon of all slots, rares of all slots, many epics.
51-55* no common weapons, cloth, leather, mail, plate armour, uncommon of many slots, rares of all slots, many epics.
56-60* no common weapons, cloth, leather, mail, plate armour, uncommon of many slots, rares of all slots, epics of most slots.
61-65* Epics of all slots. (Epics of level 46-50+ only)
66-69* Nothing (No level 51 epics)
70* Legendary Items of all major slots. Max level 50 only.

This could be under ~1,000 items or perhaps around ~500 to ~900 items. Being of ~5 item quality types spread across 50 character levels with 4 armour types and with all melee and ranged weapon types to equip and support ~3 main class / play style archetypes.

This could provide a feeling of depth and also not require too much work to create as one developer. These are just some loose ideas as to how to fill out items for a some sense of progression and depth in a video game like the Flare Empyrean Campaign. In the end you call the shots. So if your game sucks that is your fault. If your game is really good that is to your credit. For you are responsible for what goes into your game. If you don't know what is good and what is bad; get more experience and play more games and develop a bit more of your inner tastes. For when you create anything true to your heart of hearts in a way you are appealing to your inner tastes.

Everyone can do great things but you have to overcome hardships to grow as a person. Only adversity grows character. So if you have no 'character' developed inside of you it is like you can't put any 'character' into the soul of your game. This is just like writing a book. If you don't know much of anything; your book won't know much of anything either XD.

So if you feel uneasy in an area try and soak up some experience in different places and listening to people who have been there and done that, succeeded, won and beat the odds. For they alone are the best teachers you can find for they have first hand knowledge. There are many ~'fake teachers' who only read off a script that they have never lived and and personally realize from themselves. Watch out for there are many 'fake teachers' and seldom one finds real teachers; for words are cheap but actions are not. These Fake Teachers can do a lot of damage if you think third hand knowledge is the same as first hand knowledge.

So if you want to share a journey, an adventure with the world. Go and live one first. Or at least try to understand better what you like and try to soak that all up and branch out. Always try to listen to the entire world around you and strive as best you can to grow on the inside with your ~'soul' as a human being. For when you take away the all of graphics, and the sounds and the 'mental noise' of a video games you are left with nothing more than an experience waiting to be explored:-).
